### ISSUE TITLE
clang-tidy: performance-unnecessary-value-param

### ISSUE THREAD

**Issue by mbutrovich** [cmu-db/terrier#505]
We currently have this check disabled. However it should be re-enabled and the associated fixes should be applied to the codebase. It found legitimate cases where containers were being passed by value, never being mutated, and then returning - basically wasting copies when they should be references to const.

However, I would say this is currently blocked on ripping shared_ptrs out of the plan nodes which is a separate issue to revisit next month. Most of what it complains about is improper shared_ptr passing right now. Once we rip out the shared_ptrs, we should enable this check again.
